Documentation update (#602)
arkivanov/Decompose: docs/getting-started/installation.md

## The main Decompose module

The main functionality is provided by the `decompose` module. It contains some core features like:

- [ComponentContext](../component/overview.md#componentcontext) - provides API for components to manage lifecycle, state saving, etc.
- [Value](../component/overview.md/#value-and-mutablevalue-state-holders) - Multiplatform (Swift-friendly) observable state holder (allows observing state changes in the UI).
- [Child Stack](../navigation/stack/overview.md) - stack navigation model.
- [defaultComponentContext](../getting-started/quick-start.md/#android-with-compose) - creates a default `ComponentContext` attached to an Activity or Fragment on Android.
- And many other important APIs.

> This module should be imported into build.gradle.kts for shared application module
> at ```kotlin.sourceSets.commonMain.dependencies```

Reviewer: ```suggestion
This module should be imported into build.gradle.kts for shared application module at `kotlin.sourceSets.commonMain.dependencies`.
```

> If you are having Android module (androidApp or composeApp if using KMP Wizard) this dependency also needs to be
> added into build.gradle.kts for Android module at ```android.dependencies``` to allow Android code access to above functionality

Reviewer: ```suggestion
If you are having Android module (androidApp or composeApp if using KMP Wizard) this dependency also needs to be added into build.gradle.kts for Android module at ```android.dependencies``` to allow Android code access to above functionality
```

## The main Decompose module

The main functionality is provided by the `decompose` module. It contains some core features like:

> - [ComponentContext](../component/overview.md#componentcontext) needed to allow custom components
> to manage lifecycle, state saving,...

Reviewer: ```suggestion
- [ComponentContext](../component/overview.md#componentcontext) - provides API for components to manage lifecycle, state saving, etc.
```

> - [Value](../component/overview.md/#value-and-mutablevalue-state-holders) allowing UI code to observe value changes

Reviewer: ```suggestion
- [Value](../component/overview.md/#value-and-mutablevalue-state-holders) - Multiplatform (Swift-friendly) observable state holder (allows observing state changes in the UI).
```

Author: Can we somehow make it more clear that this is the class that is used within both Android and iOS/Swift? 
For example 
"Multiplatform (Swift friendly) observable state holder (allows observing state changes in the UI)."

Reviewer: Sure! Updated the suggestion.

> - [Child Stack](../navigation/stack/overview.md) navigation model

Reviewer: ```suggestion
- [Child Stack](../navigation/stack/overview.md) - stack navigation model.
```

> - [defaultComponentContext](../getting-started/quick-start.md/#android-with-compose) function for generating root component context
> that is needed in UI code

Reviewer: ```suggestion
- [defaultComponentContext](../getting-started/quick-start.md/#android-with-compose) - creates a default `ComponentContext` attached to an Activity or Fragment on Android.
```

> - and many other important things

Reviewer: ```suggestion
- And many other important APIs.
```

    Due to this fragmentation Decompose provides two separate extension modules for Compose:
    
    === "Groovy"

        ``` groovy
        implementation "com.arkivanov.decompose:extensions-compose-jetpack:<version>"
        // or
        implementation "com.arkivanov.decompose:extensions-compose-jetbrains:<version>"
        ```

    === "Kotlin"

        ``` kotlin
        implementation("com.arkivanov.decompose:extensions-compose-jetpack:<version>")
        // or
        implementation("com.arkivanov.decompose:extensions-compose-jetbrains:<version>")
        ```

>     Both modules are used to connect Compose with Decompose components. Please see the corresponding [documentation page](../extensions/compose.md).
>     Typically only one module should be selected, depending on the Compose variant being used.

Reviewer: ```suggestion
    Both modules are used to connect Compose with Decompose components. Please see the corresponding [documentation page](../extensions/compose.md). Typically only one module should be selected, depending on the Compose variant being used.
```

=== "Since v3.0.0-alpha01"

    Since Decompose version v3.0.0-alpha01 there is just one extension module for Compose - compatible with both Jetpack Compose and Multiplatform Compose.

    === "Groovy"

        ``` groovy
        implementation "com.arkivanov.decompose:extensions-compose:<version>"
        ```
    === "Kotlin"
    
        ``` kotlin
        implementation("com.arkivanov.decompose:extensions-compose:<version>")
        ```

    The module is used to connect Compose with Decompose components. Please see the corresponding [documentation page](../extensions/compose.md).

> 
> 

Reviewer: ```suggestion

```

=== "Since v3.0.0-alpha01"

    All Compose variants (Android, JVM/Desktop, Native/iOS, Native/macOS, JS/Canvas, Wasm) are published from the main branch,

>     please add dependencies as usual (no version suffixes required).

Reviewer: Unnecessary line wrap.

Author: This is also done. Sorry for marking this earlier as resolved.

!!!warning
    Wasm target is not yet supported. Please follow [issue #74](https://github.com/arkivanov/Decompose/issues/74) for more information and updates.

## Extensions for Android views

The `extensions-android` module provides extensions to connect

> Android views (older alternative to Compose UI) to Decompose components.

Reviewer: Please write single line. You can enable Soft Wrap in the IDE for better readability.

Author: Ok, will do. I did not use to Soft Wrap so will redo these.

Author: This is changed
